### PR TITLE
[Dev] Clean up of the python pkg folder structure

### PR DESCRIPTION
**Pull request by Tishj**
This PR allows use of subfolders in the pythonpkg folder, and cleans up some code in there.

Figured I'd make a PR out of this, while I work on #5407 - So no merge conflicts arise as part of these changes.

### CODE REVIEW

File: tools/pythonpkg/setup.py [hunk 1]
```diff
@@ -160,10 +160,18 @@ def __str__(self):
 extra_files = []
 header_files = []
 
+def list_source_files(directory):
+    sources = []
+    dir_len = len(directory) + len(os.sep)
+    for (subdir, dirs, files) in os.walk(directory):
+        subdir = subdir[dir_len::]
```
Comment (Mause) [inline tools/pythonpkg/setup.py:167]: Please use relpath instead of this

Or use glob to replace the entire method maybe?

**Review by Mause — CHANGES_REQUESTED** [review]

File: tools/pythonpkg/setup.py (hunk 1 above)
Comment (Mause) [reply tools/pythonpkg/setup.py:167]: My main concern is future legibility than anything

**Comment by Tishj** [comment]
Hmm so clang-tidy is failing because the generated `compile_commands.json` does not contain any commands for the pythonpkg, it gets skipped
Likely because of
```
if(NOT pybind11_FOUND)
  return()
endif()
if(NOT PythonLibs_FOUND)
  return()
endif()
```
Clang-tidy does not throw any helpful error for this though, it just tries to compile the file as is - no includes or anything

**Comment by Tishj** [comment]
Hmm I'm failing OSX Debug on my fork CI, because it's treating warnings as errors, things I fixed with
<https://github.com/duckdb/duckdb/pull/5436/commits/058249fef8ca4a6f35ebac680db112db035e603f>
But when generating the amalgamation, they still show up?
Seems related to tags/commit hashes - hopefully it's fixed once this is merged

**Comment by Mytherin** [comment]
Could you merge this with the feature branch again?

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**